.github/workflows/build.yaml

```diff
@@ -0,0 +1,197 @@
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    types: [opened, synchronize, reopened]
+  release:
+    types:
+    - created
+
+env:
+  HUGO_CONFIG: deploy.toml
+  HTMLTEST_CONFIG: .htmltest.github_actions.yml
+  YARN_BUILD_DISABLED: false
+  COMPONENTS_BUILD: true
+  CACHE_LOCATION: ./.htmltest/refcache.json
+  DOCKER_RUN_FLAGS: "--rm"
+
+jobs:
+  hugo-build:
+    name: "Build Hugo Site"
+    runs-on: ubuntu-latest
+    steps:
+    - name: "Checkout source code at current commit"
+      uses: actions/checkout@v2
+    - name: "View github event data"
+      run: cat ${GITHUB_EVENT_PATH}
+    - name: "Set up Docker Buildx"
+      uses: docker/setup-buildx-action@v1
+    - name: "Build docker image"
+      uses: docker/build-push-action@v2
+      with:
+        tags: cloudposse/docs
+        outputs: type=docker,dest=/tmp/image.tar
+    - name: "Import Image"
+      run: docker load --input /tmp/image.tar
+    - name: "Build Hugo"
+      run: |
+        set +e -x
+        make lint
+        make release
+        make real-clean hugo/build
+    - name: "Upload Docker image artifact"
+      uses: actions/upload-artifact@v2
+      with:
+        name: docker-image
+        path: /tmp/image.tar
+    - name: "Upload Hugo site artifact"
+      uses: actions/upload-artifact@v2
+      with:
+        name: hugo-site
+        path: ./public
+
+  htmltest-cache-pull:
+    name: "Pull And Update Htmltest Cache"
+    runs-on: ubuntu-latest
+    needs: hugo-build
+    steps:
+    - name: "Checkout source code at current commit"
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 2
+    - name: "Get SHA"
+      id: get_sha
+      run: |
+        # Using HEAD^# syntax to accommodate merges.
+        # It's unclear how to optimize this (i.e., whether to use HEAD^1 or HEAD^2).
+        # Perhaps a good approach is to have two versions of this step, each guarded
+        # with if statements that check the kind of event that triggered the workflow.
+        echo "::set-output name=previous_sha_short::$(git rev-parse --short HEAD^1)"
+    - name: "Download Docker image artifact"
+      uses: actions/download-artifact@v2
+      with:
+        name: docker-image
+        path: /tmp/
+    - name: "Import Image"
+      run: docker load --input /tmp/image.tar
+    - name: "Extract cache pull location"
+      id: get_cache_pull_location
+      run: |
+        export cache_pull_location=$( "${{ env.CACHE_LOCATION }}" | sed 's|\(.*\)/.*|\1|')
```

Comment:
isn't this just the `dirname`? e.g. `dirname ${{env.CACHE_LOCATION }}`

Reply:
Wow, that is a much simpler way of doing that.